src/index.ts

```diff
@@ -16,6 +17,7 @@ import { errorCatchingMiddleware, validatorMiddleware } from './validation'
 import { EmptyInputParameters } from './validation/input-params'
 
 export { FastifyInstance as ServerInstance }
+export const apiEmitter = new EventEmitter()
```

Comment:
I like the emitter approach, but let's create it in the adapter instead of making it a global var.
Additionally, let's rename it something like `shutdownNotifier` so that we clearly understand what events it's supposed to emit, imo apiEmitter is too generic